include/xbps_api_impl.h

```diff
@@ -95,6 +95,26 @@
 #define __arraycount(x) (sizeof(x) / sizeof(*x))
 #endif
 
+typedef enum transaction_file_type {
+	TYPE_LINK = 1,
+	TYPE_DIR,
+	TYPE_FILE,
+	TYPE_CONFFILE,
+} transaction_file_type_t;
+
+struct transaction_file {
+	const char *pkgname;
+	const char *pkgver;
+	char *sha256;
+	const char *target;
+	uint64_t size;
+	transaction_file_type_t type;
```

Comment:
Maybe we need two struct a `xbps_transaction_file` and a `xbps_file` which is part of the former.
The idea would be to start using the `xbps_file` struct in places where we currently use the `xbps_dictionary` references to `files.plist` so that we gradually switch from dynamic plist types to actual structs which should help to avoid some issues at compile time through the type system.
I think `xbps_file` would be a simple way to start, doing this for packages would be the end goal, there are so many places that use `xbps_dictionary`s for packages and there are a lot of places that can be improved on the way to switching to c structs.